src/MessagePack.UnityClient/Assets/Scripts/MessagePack/MessagePackSerializer.cs

```diff
@@ -462,58 +467,171 @@ private static int LZ4Operation(in ReadOnlySequence<byte> input, Span<byte> outp
+                    {
+                        int lz4Length = LZ4Operation(msgpackUncompressedData, lz4Span, LZ4CodecEncode);
 
-                    const int LengthOfUncompressedDataSizeHeader = 5;
-                    writer.WriteExtensionFormatHeader(new ExtensionHeader(ThisLibraryExtensionTypeCodes.LZ4, LengthOfUncompressedDataSizeHeader + (uint)lz4Length));
-                    writer.WriteInt32((int)msgpackUncompressedData.Length);
-                    writer.WriteRaw(lz4Span.AsSpan(0, lz4Length));
+                        const int LengthOfUncompressedDataSizeHeader = 5;
+                        writer.WriteExtensionFormatHeader(new ExtensionHeader(ThisLibraryExtensionTypeCodes.LZ4, LengthOfUncompressedDataSizeHeader + (uint)lz4Length));
+                        writer.WriteInt32((int)msgpackUncompressedData.Length);
+                        writer.WriteRaw(lz4Span.AsSpan(0, lz4Length));
+                    }
+                    finally
+                    {
+                        ArrayPool<byte>.Shared.Return(lz4Span);
+                    }
                 }
-                finally
+                else if (compression == MessagePackCompression.Lz4ContiguousBlock)
                 {
-                    ArrayPool<byte>.Shared.Return(lz4Span);
+                    // Write to [Ext(98:int,int...), bin,bin,bin...]
+                    // Reading count from Sequence<T> is better but not yet do.
+                    var sequenceCount = 0;
+                    foreach (var item in msgpackUncompressedData)
+                    {
+                        sequenceCount++;
+                    }
+
+                    writer.WriteArrayHeader(sequenceCount + 1);
+                    writer.WriteExtensionFormatHeader(new ExtensionHeader(ThisLibraryExtensionTypeCodes.LZ4Contiguous, sequenceCount * 5));
+                    {
+                        foreach (var item in msgpackUncompressedData)
+                        {
+                            // force write Int32 block(size = 5).
+                            writer.WriteInt32(item.Length);
+                        }
+                    }
+
+                    foreach (var item in msgpackUncompressedData)
+                    {
+                        var maxCompressedLength = LZ4Codec.MaximumOutputLength(item.Length);
+                        var lz4Span = ArrayPool<byte>.Shared.Rent(maxCompressedLength);
+                        try
+                        {
+                            int lz4Length = LZ4Codec.Encode(item.Span, lz4Span);
+                            writer.Write(lz4Span.AsSpan(0, lz4Length));
+                        }
+                        finally
+                        {
+                            ArrayPool<byte>.Shared.Return(lz4Span);
+                        }
+                    }
+                }
+                else
+                {
+                    throw new MessagePackSerializationException("Invalid MessagePackCompression Code.");
```

Comment:
`ArgumentException` seems the most appropriate type to throw here. And its caller wraps all exceptions in `MessagePackSerializationException` already.

Reply:
OK